app/assets/javascripts/answers.js

```diff
@@ -27,30 +27,44 @@ function validateAnswer() {
   $('.submit-answer').attr('disabled', !$('.answer-text').val().trim());
 }
 
-function initUserFeedback() {
+function initUserFeedbackQuizAll() {
   $('.btn-feedback').unbind().click(function(ev) {
     ev.preventDefault(); // prevents bootstrap from automatically adding a hidden attribute
     var $button = $(ev.target).closest('.btn-feedback');
     var feedbackActive = $button.data('feedback-active');
     var feedbackString = $button.data('feedback-string');
-    var feedbackQid = quizQuestions.questions[quizAllIndex].id;
+    var feedbackQid = quizQuestions[quizAllIndex].id;
     handleUserFeedback(feedbackActive, feedbackString, feedbackQid);
 
-    if (quizQuestions.questions.length > (quizAllIndex + 1)){
-      quizAllIndex = quizAllIndex + 1;
+    if (quizQuestions.length > (quizAllIndex + 1)){
+      quizAllIndex++;
       $('.feedback-alert').slideDown(1000, function() {
         initQuestionDisplayModalForQuizAll();
-      });
-      $('.feedback-alert').slideUp(800);
+      }).delay(1000)
+      .slideUp(800);
     }
     else {
-      $('.feedback-alert').slideDown(1000);
-      $('.feedback-alert').slideUp('slow', function(){
+      $('.feedback-alert')
+      .slideDown(1000)
```

Comment:
Let's get a `.delay(1000)` in here too, for consistency (it seems to disappear very quickly, to me).